native_lib/DDGSyncCrypto.c

```diff
@@ -184,13 +187,17 @@ DDGSyncCryptoResult ddgSyncSealOpen(
     unsigned char secretKey[DDGSYNCCRYPTO_PRIVATE_KEY_SIZE],
     unsigned char *rawBytes) {
 
-    unsigned char decrypted[cypherTextLength - crypto_box_SEALBYTES];
+    unsigned char* decrypted = new unsigned char[cypherTextLength - crypto_box_SEALBYTES];
 
+    DDGSyncCryptoResult result;
     if (crypto_box_seal_open(decrypted, cyphertext, cypherTextLength, primaryKey, secretKey) != 0) {
-        return DDGSYNCCRYPTO_SEAL_FAILED;
+        result = DDGSYNCCRYPTO_SEAL_FAILED;
+    }
+    else {
+        memcpy(rawBytes, decrypted, cypherTextLength - crypto_box_SEALBYTES);
+        result = DDGSYNCCRYPTO_OK;
     }
 
-    memcpy(rawBytes, decrypted, cypherTextLength - crypto_box_SEALBYTES);
-
-    return DDGSYNCCRYPTO_OK;
+    delete[] decrypted;
```

Comment:
@brindy @MarkIngramUK @cmonfortep I was going to use `std:unique_ptr<...>`, but then again I realized it's a C file, so I have two questions:
- Can we change it to C++? I can live with it as C
- Any good RAI approaches for C I should consider instead of my `delete[]` approach?

Reply:
Related: https://github.com/duckduckgo/sync_crypto/pull/7#discussion_r1273902511